### ISSUE TITLE
Supervisor-script prevents watchdog restart regardless of return value

### ISSUE THREAD

**Issue by janitza-luwe** [troglobit/watchdogd#41]
I tried to implement a script for a supervisor to do some logging about a process that does not met its deadline by sending a SIGUSR2 (logging is done in the programs signal handler). 
Unfortunately i disabled the watchdog this way: After calling the script, the Watchdogd did not restart the system.
Due to man page ([https://man.troglobit.com/man5/watchdogd.conf.5.html](https://man.troglobit.com/man5/watchdogd.conf.5.html)) the watchdogd should do that, as long as the supervisor-script returns non 0. Due to the log, the script _does_ return 1, but the system is not restarted.

```
23-11-06 06:53:24.641  daemon.emerg watchdogd[932]: Process foo-server[966] failed to meet its deadline, rebooting ...
23-11-06 06:53:24.649  daemon.notice watchdogd[932]: Started script /etc/watchdogd-supervisor.sh, PID 6080
23-11-06 06:53:24.775  user.notice root: script executed!
23-11-06 06:53:24.785  daemon.warn watchdogd[932]: Script (PID 6080) returned error: 1
```
Any hints?

Here is my script:
```
#!/bin/sh

if [ "$4" == "foo-server" ]; then
    if [ $(pidof foo-server) ]; then
        kill -n 12 $(pidof foo-server)
    fi
fi
logger "script executed!"
# always return non zero, otherwise watchdogd will not restart system!
exit 1
```
and my watchdogd.conf
```
supervisor {
    enabled  = true
    script = "/etc/watchdogd-supervisor.sh"
}
```
Iam working with [master head (33e32f89d4)](https://github.com/troglobit/watchdogd/commits/33e32f89d4cfe44b2d13be51b22e58eef8f7ca2f)

PS: the code looks suspicious to me. `supervisor_exec()` will always return 0. So any supervising will always be released [here](https://github.com/troglobit/watchdogd/blob/33e32f89d4cfe44b2d13be51b22e58eef8f7ca2f/src/supervisor.c#L76), if any executable script is given...

**Comment by troglobit** [comment]
Yeah you're right, that's just plain wrong.  The `supervisor_exec()` function forks the script to background, and like you say, always returns OK(0), which in turn stops the process watcher.

**Comment by troglobit** [comment]
Fixed in 6044fac and refactored in 5490655.  Verified manually in [Infix ](https://github.com/kernelkit/infix):

```
    Infix — a Network Operating System v23.12.0 (console)
    infix-00-00-00 login: admin
    Password:
    admin@infix-00-00-00:/> shell
    root@infix-00-00-00:~$ tail -F /log/syslog &
    root@infix-00-00-00:~$ watchdogctl test failed-kick
    watchdogctl: Failed kicking
    root@infix-00-00-00:~$ Dec 15 20:01:05 infix-00-00-00 watchdogd[2161]: watchdogctl[2484] tried to kick with invalid credentials: Invalid
    Dec 15 20:01:05 infix-00-00-00 watchdogd[2161]: Started script /usr/bin/supervisor-script.sh, PID 2485
    Dec 15 20:01:05 infix-00-00-00 root[2485]: process 2484 failed a kick, cause 3, restarting service.
    Dec 15 20:01:05 infix-00-00-00 root[2485]: system recovered, exit 0

    root@infix-00-00-00:~$ watchdogctl test no-kick
    Dec 15 20:01:22 infix-00-00-00 watchdogd[2161]: Started script /usr/bin/supervisor-script.sh, PID 2491
    Dec 15 20:01:22 infix-00-00-00 root[2491]: process 2490 failed to meet its deadline, cause 5
    Dec 15 20:01:22 infix-00-00-00 root[2491]: this cause is unrecoverable, exit 1!
    Dec 15 20:01:22 infix-00-00-00 watchdogd[2161]: Script (PID 2491) returned error: 1
    [ OK ] Saving system time (UTC) to RTC
    [ OK ] Saving random seed
    [ OK ] Stopping Configuration daemon
    [ OK ] Stopping NETCONF server
    [ OK ] Stopping OpenSSH daemon
    [ OK ] Stopping Status daemon
    [ OK ] Advising watchdog, system going down
    [ OK ] Unmounting filesystems ...
    [ OK ] Calling hook/svc/down scripts ...
    [ OK ] Rebooting ...
    [ OK ] watchdog: watchdog0: watchdog did not stop!
```

will be part of the v3.6 release
